source_files/watchdog.c

```diff
@@ -44,28 +46,35 @@ static void set_load(unsigned long long *loadvar, unsigned long long value)
      __asm__ __volatile__ ("mfence;");
 }
 
+pthread_t watchdog_thread;
 
 /* exit with zero returncode on sigterm */
 void sigterm_handler()
 {
     fprintf(stderr, "Caught shutdown signal, ending now ...\n");
     LOADVAR = LOAD_STOP; // required for the cases load = 100 and load = 0, which do not enter the while loop
     TERMINATE = 1;       // exit while loop used in case of 0 < load < 100
-    
+
+    pthread_kill(watchdog_thread,SIGALRM);
```

Comment:
What happens, when the `sigterm_handler` is called before `watchdog_thread` got initialized?